enterprise/views.py

```diff
@@ -552,6 +584,41 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
         return render_to_response('enterprise_course_enrollment_page.html', context_data, request=request)
 
     @method_decorator(enterprise_login_required)
+    def post(self, request, enterprise_uuid, course_id):
+        """
+        Process a submitted track selection form for the enterprise.
+        """
+        enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
+
+        selected_mode = request.POST.get('course_mode')
+
+        matching_modes = [mode for mode in modes if mode['mode'] == selected_mode]
+
+        if not matching_modes:
+            return self.get_enterprise_course_enrollment_page(request, enterprise_customer, course, modes)
+
+        if matching_modes[0].get('premium', False):
+            # Premium modes should be directed to the ecommerce flow
+            return redirect(LMS_START_PREMIUM_COURSE_FLOW_URL.format(course_id=course_id))
+
+        # Create the Enterprise backend database records for this course
+        # enrollment, then create the real enrollment in the LMS. Finally,
+        # redirect the user to the courseware page, where they'll be prompted
+        # for data sharing consent if they need to be.
```

Comment:
We should probably be using the `with transaction.atomic()` context manager here.